assets/src/block-editor/helpers/index.js

```diff
@@ -160,6 +164,60 @@ export const addAMPAttributes = ( settings, name ) => {
 	return settings;
 };
 
+/**
+ * Remove AMP deprecated props from any blocks which may contain them.
+ *
+ * @param {Object} settings Block settings.
+ *
+ * @return {Object} Modified block settings.
+ */
+export const removeDeprecatedAmpProps = ( settings ) => {
+	const mappedAttributes = {
+		ampCarousel: 'data-amp-carousel',
+		ampLightbox: 'data-amp-lightbox',
+		ampNoLoading: 'data-amp-noloading',
+		ampNoLayout: 'data-amp-layout',
+	};
+	const deprecatedProps = {};
+	const maybeBlockHasDeprecatedProp = Object.keys( mappedAttributes )
+		.some( ( attribute ) => Object.keys( settings.attributes ).includes( attribute ) );
+
+	if ( maybeBlockHasDeprecatedProp ) {
+		// At minimum, the `attributes`, and `save` props are required.
+		const deprecatedConfig = {
+			attributes: settings.attributes,
+			save( blockData ) {
```

Comment:
If `blockData.blockAttributes` lacks any of the keys in `mappedAttributes`, can this `save` function just short circuit to return `element` since at this point we know there won't be anything we need to do?

Reply:
Yes that's possible. I'll make the change.

Reply:
Actually, at that point wouldn't it already be known that `blockData.blockAttributes` would have at least 1 deprecated prop? If `maybeBlockHasDeprecatedProp` is `true` then that means at least the block would contain at least 1 deprecated prop. The next step would be to find such props, which is only possible when the block is `save`d.

Reply:
At this point we know that the block _type_ has (or _could_ have) at least one deprecated prop, but we don't know if this actual instance of that block type has the attribute.

Reply:
Ah right, I missed that. I'll include it with the fix to the issue outlined in https://github.com/ampproject/amp-wp/pull/4775#issuecomment-662782614.

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/4775/commits/d5fa7976a46c57c9ee4827dc961739c82f14ecb2.